Production/plugins/TauTupleProducer.cc

```diff
@@ -673,10 +673,10 @@ class TauTupleProducer : public edm::EDAnalyzer {
             tauTuple().ele_e2x5Max.push_back(hasShapeVars ? ele->e2x5Max() : default_value);
             tauTuple().ele_e5x5.push_back(hasShapeVars ? ele->e5x5() : default_value);
             tauTuple().ele_r9.push_back(hasShapeVars ? ele->r9() : default_value);
-            tauTuple().ele_hcalDepth1OverEcal.push_back(hasShapeVars ? ele->hcalDepth1OverEcal() : default_value);
-            tauTuple().ele_hcalDepth2OverEcal.push_back(hasShapeVars ? ele->hcalDepth2OverEcal() : default_value);
-            tauTuple().ele_hcalDepth1OverEcalBc.push_back(hasShapeVars ? ele->hcalDepth1OverEcalBc() : default_value);
-            tauTuple().ele_hcalDepth2OverEcalBc.push_back(hasShapeVars ? ele->hcalDepth2OverEcalBc() : default_value);
+            tauTuple().ele_hcalDepth1OverEcal.push_back(hasShapeVars ? ele->hcalOverEcal(1) : default_value);
```

Comment:
the code should be back-compatible with other production modes (i.e. older CMSSW releases). Please wrap include and cmssw-dependent calls. For include `__has_include` should work. You can extract CMSSW version from `PROJECT_VERSION` and then use it as a template specialisation. E.g.
```c++
namespace {
  inline constexpr int GetCMSSWVersion()
  {
    int d1 =  *(PROJECT_VERSION + 6) - '0';
    int d2 =  *(PROJECT_VERSION + 7) - '0';
    if(d2 >= 0 && d2 <= 9) return d1 * 10 + d2;
    return d1;
  }


  template<int cmssw_version>
  struct CmsswDependentImpl;

  template<>
  struct CmsswDependentImpl<10> {
    void DoSomething();
  };

  void DoSomething()
  {
    return CmsswDependentImpl<GetCMSSWVersion()>::DoSomething();
  }
}
```

Reply:
I tried a few things:
1) for instance, for single variable:
```c++
    template<int cmssw_version>
    struct FillTupleVersionDep;

    template<>
    struct FillTupleVersionDep<12> {
        float fill_electron(const pat::Electron* ele){
            return ele->hcalOverEcal(1);
        };
    };

    template<>
    struct FillTupleVersionDep<10> {
        float fill_electron(const pat::Electron* ele){
            return ele->hcalDepth1OverEcal();
        };
    };

    float get_electron(const pat::Electron* ele)
    {
        return FillTupleVersionDep<GetCMSSWVersion()>::fill_electron(ele);
    }
```
2) or just using `if conexpr (GetCMSSWVersion() > 12) {`

In both cases, it requires that this two methods of the pat::Electron i.e `ele->hcalDepth1OverEcal()` and `ele->hcalOverEcal(1)` to exist

Reply:
you need to add electron type to the template:
```c++
template<typename Ele, int cmssw_version>
struct FillTupleVersionDep;

template<typename Ele>
struct FillTupleVersionDep<Ele, 12> {
    static float fill_electron(const Ele* ele){
        return ele->hcalOverEcal(1);
    };
};

template<typename Ele>
struct FillTupleVersionDep<Ele, 10> {
    static float fill_electron(const Ele* ele){
        return ele->hcalDepth1OverEcal();
    };
};

float get_electron(const pat::Electron* ele)
{
    return FillTupleVersionDep<pat::Electron, GetCMSSWVersion()>::fill_electron(ele);
}
```

Reply:
... indeed